### PR TITLE
feat(rolling): accept zero-deployment deploys without a strategy

### PR DESCRIPTION
**Pull request by mochacat**
## Description

This allows deploys of non-Deployment objects without strategies. There is also some additional validation around manifest here.

## Motivation and Context

https://armory.atlassian.net/browse/YODL-445

## How has this been tested? How can a reviewer test these changes?

Created happy and sad path unit tests for deployment-less deploys.

# Have your performed the following tests?

Please confirm you have done the following tests to ensure your pull request is ready to be merged..

- [ ] If this change modifies the deployment flow have triggered a deployment using this branch (i.e. `build/dist/darwin_amd64/armory deploy start -f <path to file>`)?
- [ ] If this change modifes the login flow have you tried logging in and out with this branch (i.e. `build/dist/darwin_amd64/armory login`) and verified the credentials work?
- [ ] Have you verified the specific change made in this PR?

### CODE REVIEW

File: pkg/deploy/service.go [hunk 1]
```diff
@@ -266,9 +280,18 @@ func getFileNames(manifestPath model.ManifestPath) (error, []string) {
 	return err, fileNames
 }
 
-func CreateDeploymentManifests(manifests *[]string) []de.Manifest {
+func CreateDeploymentManifests(manifests *[]string, strategy *map[string]model.Strategy) ([]de.Manifest, error) {
 	deManifests := make([]de.Manifest, 0, len(*manifests))
 	for _, manifest := range *manifests {
+		var un unstructured.Unstructured
+		if strategy == nil {
+			if err := cyclopsutils.DeserializeKubernetes([]byte(manifest), &un); err != nil {
```
Comment (danielpeach) [inline pkg/deploy/service.go]: I would copy-and-paste this function into the CLI rather than depend on this part of Deploy Engine

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	github.com/ahmetb/go-linq/v3 v3.2.0
-	github.com/armory-io/deploy-engine v0.68.0
+	github.com/armory-io/deploy-engine v0.73.1
```
Comment (danielpeach) [inline go.mod]: I think the `api` package should be its own Go module so you don't have to import 100% of Deploy Engine

Comment (mochacat) [reply go.mod]: removed the cyclops dependency

**Review by aelath — APPROVED** [review]
Nice! LGTM! 🌮

**Pull request merged**